src/darwin/Framework/CHIP/MTRMetricsCollector.mm

```diff
@@ -242,7 +243,13 @@ - (void)handleMetricEvent:(MetricEvent)event
 
     // Add to the collection only if it does not exist as yet or pick latest value for instant event
     if (![_metricsDataCollection valueForKey:metricsKey] || event.type() == MetricEvent::Type::kInstantEvent) {
-        [_metricsDataCollection setValue:data forKey:metricsKey];
+        static const size_t kStagingStringLength(strlen(chip::Tracing::kMetricDeviceCommissionerCommissionStage));
+
+        // If this is the commissioning staging event, skip the cleanup to track the last stage completed in case of error
+        // For all other events, just capture the value
+        if (strncmp(event.key(), chip::Tracing::kMetricDeviceCommissionerCommissionStage, kStagingStringLength) != 0 || event.ValueUInt32() != chip::Controller::CommissioningStage::kCleanup) {
```

Comment:
This strncmp will test true for any string that starts with kMetricDeviceCommissionerCommissionStage, right?  Is that the intent? 

If that's not the intent, why are we using strncmp here at all?

Reply:
Opps, forgot to change that. Will resolve